src/devices_models/devices/common/cost_functions.jl

```diff
@@ -339,6 +347,16 @@ function ps_cost(
     # If array is full of tuples with zeros return 0.0
     all(iszero.(last.(cost_array))) && return JuMP.AffExpr(0.0)
     variable = get_variable(psi_container, var_name)[index, :]
+    settings_ext = get_ext(get_settings(psi_container))
```

Comment:
Why is this being passed with ext? There are functions to determine if the psi_container is using parameters.

Reply:
You suggested me to use psi_settings.ext instead of passing kwargs

Reply:
But you are using the ext for a field that already exists in the settings....

Reply:
I'm not using it for that, I need to know if there is a ON variable/parameter in the formulation for that device